### PR TITLE
Enable SAML realm configuration for SSO with AAD

### PR DESCRIPTION
**Pull request by russcam**
This PR enables Elasticsearch and Kibana to be configured for Single Sign-On with Azure Active Directory with the addition of two parameters:

- `samlMetadataUri`: the URI to retrieve the metadata for the Identity Provider from
- `samlServiceProviderUri`: optional URI to configure the Service Provider. This is useful where a `CNAME` DNS record for an owned domain is pointed to the canonical domain name of the Kibana public IP address. If not provided, the domain name of the Kibana public IP address will be used.

This PR builds on top of #187, so should be merged after.

The public IP address of the deployed Kibana instance must be created before the creation of the commands to execute the VM Script extensions, since the Kibana fqdn may be required as an input parameter to the commands.

A forthcoming blog post will explain what is required within Azure to enable using this feature.

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -44,42 +43,83 @@ where
...
+Which will login to the Azure account accessible by the Service Principal account and create a
+resource group with the name `test-<machinename>-<scenario>-<date>` and perform online validation
+of the template using the scenarios parameters.
+When done, the command will clean up the resource groups and logout of azure.
 
+Tests use the template checked into the github repository branch, so be sure to push any changes that need to be tested up to GitHub.
+
+### Testing a specific version
+
+By default, tests always use the last version specified in the versions array in `build/allowedValues.json`, but you can specify a version using
```
Comment (Mpdreamz) [inline build/README.md]: 👍

**Comment by Mpdreamz** [comment]
@russcam mind retargeting this PR against the branch powering #187 ? Easier to isolate the SAML changes for review :)

**Comment by russcam** [comment]
Thanks @jkakavas.

@Mpdreamz I've retargeted the PR at the branch for #187

**Comment by russcam** [comment]
Moving the creation of the kibana public ip into the mainTemplate and pushing the creation of kibana NSG and NIC into kibana-resources, allowing ubuntuSettings.json to contain all Ubuntu related settings, and node-resources.json to remain OS agnostic.

**Comment by russcam** [comment]
Would you mind taking another look please @Mpdreamz ?

File: src/scripts/elasticsearch-ubuntu-install.sh [hunk 2]
```diff
@@ -728,15 +742,16 @@ configure_elasticsearch_yaml()
     if [[ -n "${TRANSPORT_CERT}" && ${INSTALL_XPACK} -ne 0 ]]; then
       [ -d /etc/elasticsearch/ssl ] || mkdir -p /etc/elasticsearch/ssl
       log "[configure_elasticsearch_yaml] Save Transport cert blob to file"
-      echo ${TRANSPORT_CERT} | base64 -d | tee /etc/elasticsearch/ssl/elasticsearch-transport.p12
+      local TRANSPORT_CERT_PATH=/etc/elasticsearch/ssl/elasticsearch-transport.p12
```
Comment (Mpdreamz) [inline src/scripts/elasticsearch-ubuntu-install.sh]: ++ but should be part of #187

File: src/scripts/kibana-install.sh [hunk 3]
```diff
@@ -249,7 +254,7 @@ configuration_and_plugins()
         # so default to certificate mode which verifies that the provided certificate is signed
         # by a trusted authority (CA), but does not perform any hostname verification.
         echo "elasticsearch.ssl.verificationMode: certificate" >> $KIBANA_CONF
-        echo "elasticsearch.ssl.certificateAuthorities: [ \"/etc/kibana/ssl/elasticsearch-http-ca.crt\" ]" >> $KIBANA_CONF
+        echo "elasticsearch.ssl.certificateAuthorities: [ /etc/kibana/ssl/elasticsearch-http-ca.crt ]" >> $KIBANA_CONF
```
Comment (Mpdreamz) [inline src/scripts/kibana-install.sh]: should be part of #187

File: src/scripts/elasticsearch-ubuntu-install.sh (hunk 2 above)
Comment (russcam) [reply src/scripts/elasticsearch-ubuntu-install.sh]: good call, will move

File: src/scripts/kibana-install.sh (hunk 3 above)
Comment (russcam) [reply src/scripts/kibana-install.sh]: good call, will move

**Comment by russcam** [comment]
Rebased against feature/tls

**Comment by russcam** [comment]
Rebased against master

**Pull request merged**